pkg/repository/popular/registryV2/popular.go

```diff
@@ -97,5 +97,8 @@ func (p *Provider) getImageTags(img string) ([]string, error) {
 	if err = json.NewDecoder(resp.Body).Decode(&image); err != nil {
 		return nil, err
 	}
+	if !enableAllTags && len(image.Tags) > 0 {
+		return image.Tags[:1], nil
+	}
 	return image.Tags, nil
```

Comment:
Also need a test case here to assert if `enableAllTags` was false - in that case the assertion should have all tags as desired.